log/mcap.ksy

```diff
@@ -0,0 +1,397 @@
+        type: u1
+        enum: opcode
+      - id: len
+        type: u8
+
+  record:
+    seq:
+      - id: op
+        type: u1
+        enum: opcode
+      - id: len
+        type: u8
+      - id: body
+        size: len
+        type:
+          switch-on: op
+          cases:
+            opcode::header: header
+            opcode::footer: footer
+            opcode::schema: schema
+            opcode::channel: channel
+            opcode::message: message
+            opcode::chunk: chunk
+            opcode::message_index: message_index
+            opcode::chunk_index: chunk_index
+            opcode::attachment: attachment
+            opcode::attachment_index: attachment_index
+            opcode::statistics: statistics
+            opcode::metadata: metadata
+            opcode::metadata_index: metadata_index
+            opcode::summary_offset: summary_offset
+            opcode::data_end: data_end
+
+  header:
+    seq:
+      - id: profile
+        type: prefixed_str
+      - id: library
+        type: prefixed_str
+
+  footer:
+    seq:
+      - id: summary_start
+        type: u8
+      - id: summary_offset_start
+        type: u8
+      - id: summary_crc
+        type: u4
+    instances:
+      summary_section:
+        io: _root._io
+        pos: summary_start
+        size: "(summary_offset_start != 0 ? summary_offset_start : _root._io.size - sizeof<magic> - sizeof<footer> - sizeof<record_prefix>) - summary_start"
+        type: records
+        if: summary_start != 0
+      summary_offset_section:
+        io: _root._io
+        pos: summary_offset_start
+        size: "_root._io.size - sizeof<magic> - sizeof<footer> - sizeof<record_prefix> - summary_offset_start"
+        type: records
+        if: summary_offset_start != 0
+
+  schema:
+    seq:
+      - id: id
+        type: u2
+      - id: name
+        type: prefixed_str
+      - id: encoding
+        type: prefixed_str
+      - id: len_data
+        type: u4
+      - id: data
+        size: len_data
+
+  channel:
+    seq:
+      - id: id
+        type: u2
+      - id: schema_id
+        type: u2
+      - id: topic
+        type: prefixed_str
+      - id: message_encoding
+        type: prefixed_str
+      - id: metadata
+        type: map_str_str
+
+  message:
+    seq:
+      - id: channel_id
+        type: u2
+      - id: sequence
+        type: u4
+      - id: log_time
+        type: u8
+      - id: publish_time
+        type: u8
+      - id: data
+        size-eos: true
+
+  chunk:
+    seq:
+      - id: message_start_time
+        type: u8
+      - id: message_end_time
+        type: u8
+      - id: uncompressed_size
+        type: u8
+      - id: uncompressed_crc
+        type: u4
+      - id: compression
+        type: prefixed_str
+      - id: len_records
+        type: u8
+      - id: records
+        size: len_records
+        type:
+          switch-on: compression.str
+          cases:
+            '""': records
+
+  message_index:
+    types:
```

Comment:
`seq` should come before `types`, see our [style guide](https://doc.kaitai.io/ksy_style_guide.html).